### PR TITLE
Homepage special projects secondary nav

### PR DESCRIPTION
**Pull request by joshdarby**
## Changes

This pull request makes the following changes:

- Adds the secondary nav menu to `nav-main.php` if on the homepage
- Uses the `wp_nav_menu_objects` filter to create a function to find featured media for each menu item
- Adds menu styling for this specific menu

Normal test with only series and categories:
Screen Shot 2019-10-22 at 4 45 46 PM

Test with a custom link menu item with no image:
Screen Shot 2019-10-22 at 4 46 37 PM

Test with series, categories, and posts:
Screen Shot 2019-10-22 at 4 47 23 PM

Test with series, categories, and a post with no featured image:
Screen Shot 2019-10-22 at 4 47 54 PM

## Why

For #41 

## Testing/Questions

Features that this PR affects:

- Secondary special projects nav

Questions that need to be answered before merging:

- [x] Is this PR targeting the correct branch in this repository?
- [x] What type of object did I miss to grab featured media from?
- [x] Is there a better way to grab and display featured media for each item?

Steps to test this PR:

1. View the special nav on the frontend
2. Experiment with adding different types of menu items

### CODE REVIEW

File: wp-content/themes/citylimits/less/_navbar.less [hunk 1]
```diff
@@ -290,4 +290,53 @@ html.nav-open body {
...
         display: none;
     }
+}
+#secondary-nav {
+    .secondary-special-projects-nav {
+        list-style: none;
+        margin-left: 0;
+        margin-top: -5px;
+        display: flex;
+        align-items: center;
```
Comment (benlk) [inline wp-content/themes/citylimits/less/_navbar.less]: If entries with different numbers of lines don't all have images, then the border-right is of different heights:

Screen Shot 2019-10-22 at 18 22 12

Do we care if that is the case, or is the operating assumption that they'll all have images?

**Comment by benlk** [comment]
Do you have "Series" in the menu-edit window?

Screen Shot 2019-10-22 at 18 20 40

File: wp-content/themes/citylimits/less/_navbar.less [hunk 2]
```diff
@@ -290,4 +290,53 @@ html.nav-open body {
...
+        background: @red;
+        li {
+            flex-grow: 1;
+            flex-basis: 0;
+            margin: 20px 0;
+            border-right: 1px solid @grey1;
+            padding-left: 5%;
```
Comment (benlk) [inline wp-content/themes/citylimits/less/_navbar.less]: Should this padding be replicated on the right? As it stands, it looks kinda tossed-together: Screen Shot 2019-10-22 at 18 26 41


Should it be reduced to `@measure`? In the below screenshot, `@measure`'s `1.5rem` replaces the `2%`.

Screen Shot 2019-10-22 at 18 28 39

**Comment by benlk** [comment]
>  What type of object did I miss to grab featured media from?

Custom links, but that's a rabbithole that I don't think we need to support.

> Is there a better way to grab and display featured media for each item?

This looks fine. It's simpler than https://github.com/INN/umbrella-invwest/blob/6ed273b187892664c208091a540f8d8376c53773/wp-content/themes/invw/homepages/zones/invw-zones.php#L163-L235 was.

**Comment by joshdarby** [comment]
> Do you have "Series" in the menu-edit window? It's not there in mine.

@benlk Yep, I can see it on my local environment and on staging

Screen Shot 2019-10-23 at 8 52 40 AM

File: wp-content/themes/citylimits/less/_navbar.less (hunk 1 above)
Comment (joshdarby) [reply wp-content/themes/citylimits/less/_navbar.less]: I _think_ the assumption is that they'll all have images, but I'll update it to make sure all of the list items get the same height so their border heights are the same.

Comment (joshdarby) [reply wp-content/themes/citylimits/less/_navbar.less]: @benlk https://github.com/INN/umbrella-citylimits/pull/78/commits/5f9627bc08354b28f70805d6edc58164b0cb1eaa updates it so the list items are all the same height, giving them all the same height border

File: wp-content/themes/citylimits/less/_navbar.less (hunk 2 above)
Comment (joshdarby) [reply wp-content/themes/citylimits/less/_navbar.less]: @benlk https://github.com/INN/umbrella-citylimits/pull/78/commits/79610c5e9b67fe13113c273e46d440073db9b52d uses `@measure` for the left padding and adds it to the right as well

**Comment by benlk** [comment]
Hm. I wonder why I'm not seeing the series, neighborhoods, or other taxonomies in my dashboard.

**Pull request merged**